crates/pixi_build_frontend/src/tool/installer.rs

```diff
@@ -0,0 +1,276 @@
+use std::fmt::Debug;
+use std::future::Future;
+use std::path::PathBuf;
+
+use pixi_consts::consts::{CACHED_BUILD_ENVS_DIR, CONDA_REPODATA_CACHE_DIR};
+use pixi_progress::wrap_in_progress;
+use pixi_utils::{EnvironmentHash, PrefixGuard};
+use rattler::{install::Installer, package_cache::PackageCache};
+use rattler_conda_types::{Channel, ChannelConfig, GenericVirtualPackage, Platform};
+use rattler_repodata_gateway::Gateway;
+use rattler_shell::{
+    activation::{ActivationVariables, Activator},
+    shell::ShellEnum,
+};
+use rattler_solve::{resolvo::Solver, SolverImpl, SolverTask};
+use rattler_virtual_packages::{VirtualPackage, VirtualPackageOverrides};
+use reqwest_middleware::ClientWithMiddleware;
+
+use super::{
+    cache::ToolCache, IsolatedTool, IsolatedToolSpec, SystemTool, Tool, ToolCacheError, ToolSpec,
+};
+
+use miette::{miette, IntoDiagnostic};
+
+/// A trait that is responsible for installing tools.
+pub trait ToolInstaller {
```

Comment:
~~What is the point of this trait as I see only one implementation?~~

Edit: At first I was like, why?? But then I saw the test and it convinced me.